docs/source/routing/self-hosted/containerization/gcp.mdx

```diff
@@ -0,0 +1,161 @@
+    - Save your `APOLLO_KEY` and `APOLLO_GRAPH_REF` in your GCP Secret Manager. You'll need them when deploying the router.
+1. Install [Docker](https://www.docker.com/get-started/) locally.
+1. Create a [GCP account](https://cloud.google.com/) or use an existing account.
+1. Create a [GCP project](https://console.cloud.google.com/projectcreate). Choose a project name (for example, `my-project`) and save it to use later when deploying the router.
+1. Install the [gcloud CLI](https://cloud.google.com/sdk/docs/install) and log in to your GCP account.
+1. Choose a version of the router to deploy (for example, `v1.61.0`). You'll need it when specifying the router image to deploy.
+
+## Build router image
+
+To deploy your own router, start by customizing and building a router image, using a Dockerfile and a router configuration file:
+
+1. In a local directory, create a `router.yaml` file and copy-paste the following configuration into the file: 
+    
+    ```yaml title="router.yaml"
+    supergraph:
+        listen: 0.0.0.0:4000
+    health_check:
+        listen: 0.0.0.0:8088
+    ```
+
+    The router's default HTTP and health check endpoint addresses are localhost, so they wouldn't be reachable when deployed. This configuration enables the router to listen to all addresses.
+
+1. Create a `Dockerfile` file and copy-paste the following into the file:
+
+    ```text showLineNumbers=false
+    # Use the official Apollo Router Core image as the base.
+    # Set the image tag to the desired router version (e.g. v1.61.0)
+    FROM ghcr.io/apollographql/router:v1.61.0
+
+    # Replace the default router config with the local, customized router.yaml
+    COPY router.yaml /dist/config/router.yaml
+    ```
+
+    The Dockerfile sources the base router image from the GitHub container registry, using the version of router you specify. It then copies your customized `router.yaml` configuration file to overwrite the default router configuration file.
+
+1. From the same local directory, use `docker buildx` CLI command to build a new router image. Choose a name and tag for the image, for example `router-gcp:v1.61.0`.
+    
+    ```bash showLineNumbers=false
+    docker buildx build --platform linux/amd64  -t router-gcp:v1.61.0 --load .
+    ```
+
+    - Because Cloud Run only supports AMD64-based images, the `docker buildx build --platform linux/amd64` command ensures the image is built for AMD64 and is compatible.  
+    - The `--load` option loads the built image to `docker images`.
+
+1. Use `docker images` to validate that your router image is successfully built and loaded.
+
+## Push router image to container registry
+
+With a router image built, set up GCP Artifact Registry, then tag and push your image to it.
+
+### Set up container registry
+
+1. In GCP, enable [Artifact Registry](https://cloud.google.com/artifact-registry/docs/enable-service) in your project.
```

Comment:
Consider simply using a paragraph for a single-step procedure instead of an ordered list.